### PR TITLE
refactor(ivy): remove style sanitization

### PR DESCRIPTION
**Pull request by matsko**
This patch is the first of many commits to disable sanitization for
[stlye.prop] and [style] bindings in Angular.

Historically, style-based sanitization has only been required for old
IE browsers (IE6 and IE7). Since Angular does not support these old
browsers at all, there is no reason for the framework to support
style-based sanitization.

### CODE REVIEW

**Review by IgorMinar — APPROVED** [review]
one q, but otherwise LGTM! thanks

File: packages/core/src/sanitization/sanitization.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export function ɵɵsanitizeStyle(unsafeStyle: any): string {
   if (allowSanitizationBypassAndThrow(unsafeStyle, BypassType.Style)) {
     return unwrapSafeValue(unsafeStyle);
   }
-  return _sanitizeStyle(renderStringify(unsafeStyle));
+  return renderStringify(unsafeStyle);
```
Comment (kara) [inline packages/core/src/sanitization/sanitization.ts:72]: Can we update the docs comment above? It seems to be outdated now that we no longer remove urls/javascript.

It looks like we also should update `defaultStyleSanitizer`

**Review by kara — CHANGES_REQUESTED** [review]
Thanks! Some comments below

File: aio/content/guide/deprecations.md [hunk 2]
```diff
@@ -517,6 +517,10 @@ The final decision was made on three key points:
 * __Usage statistics__: We looked at usage trends for IE 9 and 10 from various sources and all indicated that usage percentages were extremely small (fractions of 1%).
 * __Feedback from partners__: We also reached out to some of our Angular customers and none expressed concern about dropping IE 9 and 10 support.
 
+{@a style-sanitization}
+### Style Sanitization for `[style]` and `[style.prop]` bindings
```
Comment (kara) [inline aio/content/guide/deprecations.md:521]: Move to removed apis section

File: aio/content/guide/deprecations.md [hunk 3]
```diff
@@ -526,6 +530,7 @@ The following APIs have been removed starting with version 10.0.0*:
 | ---------------- | -------------- | ----------- | ----- |
 | `@angular/core`  | Undecorated base classes that use Angular features | Add Angular decorator | See [migration guide](guide/migration-undecorated-classes) for more info |
 | `@angular/core`  | `ModuleWithProviders` without a generic             | `ModuleWithProviders` with a generic | See [migration guide](guide/migration-module-with-providers) for more info |
+| `@angular/core`  | Style Sanitization | no action needed | Sanitization for `[style]` and `[style.prop]` bindings |
```
Comment (kara) [inline aio/content/guide/deprecations.md:533]: Link to notes

**Review by kara — CHANGES_REQUESTED** [review]
A few more quick things

**Review by kara — APPROVED** [review]
LGTM

**Comment by matsko** [comment]
Merge assistance needed. This PR conflicts with a patch within G3 (`g3_patches/disable_style_sanitization.patch`) and cannot be merged in normally. What needs to happen first is:

1) The patch marked above needs to be removed in a single CL
2) This PR needs to be synced normally.

This might be possible by creating a client that deletes the patch together with the PR code. I will give this a shot.

**Comment by matsko** [comment]
Here's a presubmit CL:  cl/310224253

**Pull request closed without merging**